src/ILCompiler.ReadyToRun/src/Compiler/ReadyToRunTableManager.cs

```diff
@@ -4,21 +4,46 @@
 
 using System;
 using System.Collections.Generic;
+using System.Reflection.Metadata;
+using System.Reflection.Metadata.Ecma335;
 
 using ILCompiler.DependencyAnalysis;
 using ILCompiler.DependencyAnalysis.ReadyToRun;
 using ILCompiler.DependencyAnalysisFramework;
 
 using Internal.TypeSystem;
+using Internal.TypeSystem.Ecma;
 
 using Debug = System.Diagnostics.Debug;
 
 namespace ILCompiler
 {
-    public class ReadyToRunTableManager : MetadataManager
+    public struct DefinedTypeInfo
```

Comment:
This could be a generic struct where the T is either `TypeDefinitionHandle` or `ExportedTypeHandle` to save a dozen lines of code.

Reply:
Good point, will fix.

Reply:
Fixed in 4th commit.